test/web-audio-playerSpec.js

```diff
@@ -1,77 +1,476 @@
-            });
-          } else {
-            handler();
-          }
+              }
+            };
+          errorCodes.forEach(code => {
+            parameter.target.error.code = code;
+            handler(parameter);
+          });
+        } else {
+          handler();
         }
       };
+      var options = jasmine.createSpyObj('options', [
+        'playingCb',
+        'timeupdateCb',
+        'durationchangeCb',
+        'canplayCb',
+        'endedCb',
+        'pauseCb',
+        'progressCb',
+        'errorCb'
+      ]);
+      webAudioPlayer = new WebAudioPlayer(options);
+      webAudioPlayer._pauseIsStop = true;
+      audioMock.firePausedEvent();
 
-    spyOn(window, 'Audio').and.returnValue(audioMock);
-    spyOn(console, 'error');
-
-    var options = jasmine.createSpyObj('options', [
-      'playingCb',
-      'timeupdateCb',
-      'durationchangeCb',
-      'canplayCb',
-      'endedCb',
-      'pauseCb',
-      'progressCb',
-      'errorCb'
-    ]);
-
-    var webAudioPlayer = new WebAudioPlayer(options);
-    webAudioPlayer._pauseIsStop = true;
-    audioMock.firePausedEvent();
-
-    expect(options.playingCb).toHaveBeenCalledTimes(1);
-    expect(options.timeupdateCb).toHaveBeenCalledTimes(1);
-    expect(options.durationchangeCb).toHaveBeenCalledTimes(1);
-    expect(options.canplayCb).toHaveBeenCalledTimes(1);
-    expect(options.endedCb).toHaveBeenCalledTimes(1);
-    expect(options.pauseCb).toHaveBeenCalledTimes(1);
-    expect(options.progressCb).toHaveBeenCalledTimes(1);
-    expect(options.errorCb).toHaveBeenCalledTimes(5);
-
-    expect(console.error).toHaveBeenCalledTimes(5);
-    expect(console.error).toHaveBeenCalledWith('You aborted the playback.');
-    expect(console.error).toHaveBeenCalledWith('A network error caused the audio download to fail.');
-    expect(console.error).toHaveBeenCalledWith('The audio playback was aborted due to a corruption ' +
-      'problem or because the media used features your browser did not support.');
-    expect(console.error).toHaveBeenCalledWith('The audio could not be loaded, either because the ' +
-      'server or network failed or because the format is not supported.');
-    expect(console.error).toHaveBeenCalledWith('An unknown error occurred.');
+      expect(webAudioPlayer.settings).toEqual(options);
+      expect(webAudioPlayer._pauseIsStop).toBeFalsy();
+      expect(options.playingCb).toHaveBeenCalledTimes(1);
+      expect(options.timeupdateCb).toHaveBeenCalledTimes(1);
+      expect(options.durationchangeCb).toHaveBeenCalledTimes(1);
+      expect(options.canplayCb).toHaveBeenCalledTimes(1);
+      expect(options.endedCb).toHaveBeenCalledTimes(1);
+      expect(options.pauseCb).toHaveBeenCalledTimes(1);
+      expect(options.progressCb).toHaveBeenCalledTimes(1);
+      expect(options.errorCb).toHaveBeenCalledTimes(5);
+      expect(console.error).toHaveBeenCalledTimes(5);
+      expect(console.error).toHaveBeenCalledWith('You aborted the playback.');
```

Comment:
I wonder if there's a better way to parameterize the tests in JS like pytest in python allows